prow/cmd/splice/main.go

```diff
@@ -202,6 +202,24 @@ func (s *splicer) makeBuildRefs(org, repo string, prs []int) kube.Refs {
 	return refs
 }
 
+func requiredJobs(presubmits []config.Presubmit, refs kube.Refs, succeeded map[string]bool) []config.Presubmit {
+	var out []config.Presubmit
+	for _, job := range presubmits {
+		if !job.AlwaysRun { // Ignore manual jobs as these do not block
+			continue
+		}
+		if job.SkipReport { // Ignore silent jobs as these do not block
+			continue
+		}
+		if succeeded[refs.String()+job.Context] {
```

Comment:
I think this shouldn't be pulled into this function. It makes the function feel less pure, since the context for how succeeded is built and referenced (ref + context) is in main().